lib/colorUtil.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:avf/model.dart' as model;
+
+// todo: this is temporary, this will be moved to server
+List<List<num>> _getThemeColors(List<model.Theme> themes) {
```

Comment:
I like that you've split the work of giving a color to each theme into two parts. I don't think actually at this point you need to pass on the list of themes, you only need the minimum number of themes to be generated, right? So the parameter can be just an integer.

The name is also a bit misleading. How about '_generateColorsPool`? (I'm not 100% sure about "pool" though, maybe "collection" would be better? Basically to indicate that you're actually generating a surplus of colors from which you will chose later on).

As a side question - why is this temporary? Colours are primarily a presentation issue - I'm not sure it's a good idea to mix data (what's on the server) with what's in the UI. Is it because you want to give people the option to set the colours manually? And how likely is that to be needed? (Just trying to understand a bit the context and think through the options, if this needs to be on the server, that's fine!)

Reply:
Ah I missed seeing it as I only use themes.length anyway. Fixed.

I think collection is better since I assign them in the order of it being generated. 

What you suggest makes sense - but there are a couple of constraints had to be put in place. 1. The colours assigned to each category should not change thorough the lifecycle of the chart and 2. that the colours don't clash (similar colours next to each other) on the bar graph. We currently sort the chart by theme frequency.

So as we keep adding more data into people (or adding labels to existing people), the frequency of the themes will change and it is not possible to know the sort order of themes in the client without requesting all the people first. So it made sense to add this algorithm to the db write script as we write into people / themes to assign an order as well as the colour to the theme.

Is this a sensible thing to do? Or are there other simpler ways to think about this?

Reply:
Yeah, this is a tricky one. I'm guessing then that this is going to be a live dashboard thing, right?
In which case, assigning the colour in the backend makes sense. You could also assign just the order in the backend and set the colours in the UI based on that order, but it doesn't make too much difference really. Either way would be fine really.

Reply:
That is true - all I need is the order which I already have in the db. It doesn't make any difference for the colours where it is done, will move to the client.